### PR TITLE
EE-625 Add EGA Policy to metadata API

### PR DESCRIPTION
**Pull request by Zhicheng-Liu**

### CODE REVIEW

File: src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/idconverter/DbIdVersionEntityIdConverter.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    @Override
+    public DbIdVersionEntityId convert(String id) {
+        DbIdVersionEntityId dbIdVersionEntityId;
+        try {
+            String[] DbIdVersion = id.split("\\.");
+            String[] DbId = DbIdVersion[0].split("\\_");
+            dbIdVersionEntityId = new DbIdVersionEntityId(DbId[0], DbId[1], DbIdVersion[1]);
+        } catch (Exception ex) {
+            throw new IllegalArgumentException("Please provide an ID in the form accession.version");
```
Comment (selvaebi) [inline src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/idconverter/DbIdVersionEntityIdConverter.java]: should be db_id.version.

File: src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/DbIdVersionEntityId.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+    @Size(min = 1, max = 32)
+    @NotNull
+    private String db;
+
+    @Size(min = 1, max = 32)
+    @NotNull
+    private String id;
+
+    @Size(min = 1, max = 32)
```
Comment (selvaebi) [inline src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/DbIdVersionEntityId.java]: display default version to be 1 in swagger

File: src/test/java/uk/ac/ebi/ampt2d/metadata/MetadataApplicationTest.java [hunk 3]
```diff
@@ -1448,4 +1471,169 @@ public void findDacByAccession() throws Exception {
...
+        mockMvc.perform(get(duoUrl))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$..self.href").value(duoUrl));
+
+        mockMvc.perform(get(duoUrl + "/condition"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$..self.href").value(crUrl));
+
+        String modifierUrl = postTestCrossReference("DUO", "0000008");
+        String duoWithModifierUrl = postTestDuo(crUrl, Arrays.asList(modifierUrl));
```
Comment (selvaebi) [inline src/test/java/uk/ac/ebi/ampt2d/metadata/MetadataApplicationTest.java]: modification should be done by patch method, so if u kinda add auto id this should fail.

File: src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/idconverter/DbIdVersionEntityIdConverter.java (hunk 1 above)
Comment (Zhicheng-Liu) [reply src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/idconverter/DbIdVersionEntityIdConverter.java]: Fixed.

File: src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/DbIdVersionEntityId.java (hunk 2 above)
Comment (Zhicheng-Liu) [reply src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/DbIdVersionEntityId.java]: Fixed.

File: src/test/java/uk/ac/ebi/ampt2d/metadata/MetadataApplicationTest.java (hunk 3 above)
Comment (Zhicheng-Liu) [reply src/test/java/uk/ac/ebi/ampt2d/metadata/MetadataApplicationTest.java]: No, this is not a modification to a created object. The `Duo` object happens to have a field called `modifiers` which is used optionally to further specifiy the field `condition`.  Here `Duo`s with same `condition`s but with different `modifiers` are treated as different objects.

**Comment by Zhicheng-Liu** [comment]
I have added auto generated IDs and unique constraints for those EGA objects.

**Pull request merged**